src/db.c

```diff
@@ -1112,8 +1114,17 @@ void scanGenericCommand(client *c, robj *o, unsigned long cursor) {
         while ((ln = listNext(&li))) {
             sds key = listNodeValue(ln);
             initStaticStringObject(kobj, key);
-            if (expireIfNeeded(c->db, &kobj, 0)) {
-                listDelNode(keys, ln);
+            /* Filter an element if it isn't the type we want. */
+            if (typename) {
+                robj* typecheck = lookupKeyReadWithFlags(c->db, &kobj, LOOKUP_NOTOUCH|LOOKUP_NONOTIFY);
```

Comment:
@oranagra In order to ensure that static robj is supported in lookupKey, I added the NONOTIFY flag here to avoid modifying the notifyKeyspaceEvent function.  I think the `keymiss` notification generated by the scan command is completely unnecessary. But I am not sure whether it will also be a breaking change.

Reply:
i agree.. in theory this should have been a `LOOKUP_NOEFFECTS`, or actually just a plain dictFind, but currently we do want the expiration side effect.
regarding the `keymiss` notification, if SCAN would ever send them, that would be a bug (badly affecting whoever is using relies on this notification), and in such case i would document that in the release notes as a bug fix.
but in fact, these should always be hits (never misses), so i don't think that bug exists.
what you did is ok, and it's a good idea to keep NONOTIFY just to be on the safe side.